### DISCUSSION TITLE
How to use External Config?

### DISCUSSION THREAD

**Discussion by erfjab** [Gozargah/Marzban#1181] [General]
اول از همه اصلاً این External Config چیه؟ با این قابلیت شما می‌توانید یک کانفیگ ویتوری که در پنل شما ساخته نشده را وارد ساب‌های کاربرانتان کنید. مثلاً در مواقع قطعی و اختلال مجبورید یک کانفیگ موقتی از پنل دیگه بالا بیاورید و به همه دونه دونه کاربران ارسال کنید یا اگر یک کانفیگ جدید دارید که می‌خواهید کاربران از آن استفاده کنند، با این قابلیت می‌توانید به راحتی آن کانفیگ را وارد ساب همه کاربران خود کنید.

برای استفاده از این قابلیت، ابتدا وارد فایل env شوید:
```
nano /opt/marzban/.env
```
سپس این متغیر را با کانفیگ خود اضافه کنید:
```
EXTERNAL_CONFIG='config://....'
```
و در نهایت مرزبان خود را ریستارت کنید:
```
marzban restart
```
و تمام.

دقت کنید که در حال حاضر این کانفیگ فقط برای ساب پیچ‌ها و نرم‌افزارهایی که کانفیگ فرمت v2ray دریافت می‌کنند کار می‌کند. یعنی اگر از کانفیگ‌های کاستوم استفاده کنید، این قابلیت برای شما کار نخواهد کرد. چرا؟ چون این قابلیت هنوز در مراحل ابتدایی توسعه است و قصد دارم آن را به گونه‌ای توسعه دهم که تمامی شرایط را پوشش دهد. 

چون این کانفیگ در پنل شما ساخته نشده، مصرف کاربران از حجمشان کم نمی‌شود.

و بله، فقط می‌توانید یک کانفیگ اضافه کنید.

**Comment by erfjab** [comment]
> عشقی. خیلی خوبی خیلی. 
> اگه یه آموزش هم تو کانالت بزاری اینکه چطوری بدون بوجود اومدن مشکل تو دیتابیس Mysql از نسخه 0.4.9 مرزبان به نسخه جدید آپدیت کنیم عالی میشه، چون جایی ندیدم بزارن.

قربونت داداش، مرزبان نامه گذاشته بود که. این پست رو چک کنید.
https://t.me/Marzban_Nameh/262

**Reply by erfjab** [reply]
> نمیدونم چجوری از زحماتی که بابت مرزبان و آن‌هولد میکشید تشکر کنم. امیدوارم هرجا که هستی و هرکاری که میکنی موفق و خوشبخت باشی عزیز دل.

از جملات پرلطف‌تون خیلی ممنونم عزیز. من فقط یه گوشه‌ی کار رو دست میندازم. ♥️🙏

**Comment by kiaan98** [comment]
مهندس جان دس طلا.    من همین خط رو اضافه کردم و بجای کلمه کانفیگ ، کانفیگئ گذاشتم مرزبانم ریستارت کردم ولی نیومد تو لیست سابم  اکسترنال کانفیگ.میشه راهنمایی کنین

اینجوری زدم من

EXTERNAL_CONFIG='کانفیگ من'

**Reply by erfjab** [reply]
> مهندس جان دس طلا.    من همین خط رو اضافه کردم و بجای کلمه کانفیگ ، کانفیگئ گذاشتم مرزبانم ریستارت کردم ولی نیومد تو لیست سابم  اکسترنال کانفیگ.میشه راهنمایی کنین
> 
> اینجوری زدم من
> 
> EXTERNAL_CONFIG='کانفیگ من'

درود و وقت بخیر، راجب اینکه در چه شرایطی نمیاد توضیحاتی دادم خدمتتون، در حال حاضر فقط برای کانفیگ v2ray (بدون کاستوم) میاره. چون حجم وسیعی از کدنویسی نیاز بود، تصمیم گرفتیم قدم به قدم توسعه بدیم.
